### ISSUE TITLE
Could you provide the training file?

### ISSUE THREAD

**Issue by Joooooohnny** [GewelsJI/PNS-Net#5]
I encounter a problem when I am trying to implement your work on my own(video shadow segmentation). The Normalized Self-attention block does not work as you do.

**Comment by GewelsJI** [comment]
Hi,

Thanks for your attention to our work.

We are planing to extend this work to journal version, and thus, the training code will be public in a near future. Besides, what's your problem encountered?

Best regards,
Ge-Peng.

**Comment by Joooooohnny** [comment]
Hi,
Thank you for your reply.
The performance of the model I built get worse when I am trying to add Normalised Self-Attention Block. I tried to fine-tune the NSB hyper-parameters, but it did not work. What’s more, what loss function do you use for training? I tried the cross-entropy loss you mentioned on your paper. But it did not work either. I think something wrong with my own implementation, but I am having a hard time finding it out.

Best regards,
Johnny.

**Comment by GewelsJI** [comment]
Hi,

We use this loss function in our implementation:

```python
class CrossEntropyLoss(nn.Module):
    def __init__(self):
        super(CrossEntropyLoss, self).__init__()

    def forward(self, *inputs):
        pred, target = tuple(inputs)
        total_loss = F.binary_cross_entropy(pred.squeeze(), target.squeeze().float())
        return total_loss
```

In my own experience, you can turn down the learning rate in the fine-tuning phase. The different tasks may incur different problems, but I think this core module could also work well in the video shadow segmentation task. Thus, you can make many attempts on tuning the hyper-parameters. If it still does not work, I recommend you can freeze or slow down the feature extraction layers.

Best regards,
Ge-Peng.

**Comment by Joooooohnny** [comment]
Hi,

Thank you for your help!!!
I will make a try.

Best regards,
Johnny

**Comment by GewelsJI** [comment]
UR welcome. Keep in touch with me if you have any progress on it. 

I will close this issue temporally.

Best regards,
Ge-Peng.
